### PR TITLE
Add ability to declare Solr object with default commit policy

### PR DESCRIPTION
**Pull request by efagerberg**
Since solr has hooks to auto commit after X documents or X seconds, it is best that the `Solr` methods `get`, `delete`, and `_update` do not have a policy of committing by default.

One may already assume this to be false by default since it is good practice to limit commits. But incase users rely on this functionality, there is the ability to declare the default policy in the `Solr` constructor.

### CODE REVIEW

**Comment by efagerberg** [comment]
@acdha you mind giving this a look over?

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
 .tox
 solr*.tgz
 solr-app
-solr
```
Comment (acdha) [inline .gitignore]: This change is unrelated

File: pysolr.py [hunk 2]
```diff
@@ -898,11 +905,12 @@ def add(self, docs, boost=None, fieldUpdates=None, commit=True, softCommit=False
         m = force_unicode(m)
 
         end_time = time.time()
+        commit = self._get_commit(commit)
```
Comment (acdha) [inline pysolr.py]: Do we really need to make this call here when it's just calling `_update()` which does the same check? I'm thinking we could remove it in both `_add` & `_delete` and inline the `_get_commit` check into `_update`.

**Comment by efagerberg** [comment]
Thanks I will review this sometime tomorrow.

**Comment by acdha** [comment]
Thanks for sending the request - it's a good idea but this has been a busy period at work for me to get around to OSS reviews.

**Comment by efagerberg** [comment]
@acdha you were totally right about `_get_commit`. Fixed that up and added a those tests.

**Comment by acdha** [comment]
Can you rebase this against the current master branch & confirm that the tests still pass?

File: README.rst [hunk 3]
```diff
@@ -156,6 +156,25 @@ with: ``mlt``, ``update``, ``terms`` etc. The specific methods of pysolr's ``Sol
...
 
 
+Custom Commit Policy
+~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    # Setup a Solr instance. The trailing slash is optional.
+    # All request to solr will result in a commit
+    solr = pysolr.Solr('http://localhost:8983/solr/core_0/', search_handler='/autocomplete', commit_by_default=True)
```
Comment (mlissner) [inline README.rst]: Maybe this should be something like send_commits or something? by_default doesn't sound right to me.

**Comment by efagerberg** [comment]
@acdha sorry about the wait on this, rebased onto master. @mlissner I have changed the parameter name to `always_commit` is that more appropriate?

**Comment by mlissner** [comment]
And this should not be closed either. Can't say I'm a fan of the stale bot.

**Comment by acdha** [comment]
Thanks for updating it. It looks good to me. The only question I had is whether we should default to `always_commit=True` for backwards-compatibility on the assumption that people who aren't pressed on are less likely to read the documentation but I don't feel especially strongly about it.

**Comment by mlissner** [comment]
I think we concluded that switching to `always_commit=False` was better because it:

 - would be a good thing for most people, who don't realize that commits should be infrequent in Solr.
 - would be a good reason to do a major version bump
 - should have minimal impact because of Solr's default commit policy (which regularly does commits)

**Comment by mlissner** [comment]
Oh, and I do feel pretty strongly about this, because always_commit=True is a weird default that'd definitely mess some people up.

**Comment by acdha** [comment]
Fine by me

**Comment by efagerberg** [comment]
Thanks folks 😄

**Comment by acdha** [comment]
@efagerberg if you want to draft that wording in a pull request it's a good idea.
